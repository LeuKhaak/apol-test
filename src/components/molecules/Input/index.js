import React from "react";
import * as styles from "./styles.module.scss";

const Input = ({ userId, el, ind, value, edit, saveRowDataForInput }) => {
  return (
    <div className={styles.inputWrapper}>
      <input
        id={userId + el[0]}
        name={el[0]}
        value={value}
        onChange={(event) => edit(ind, event)}
        onKeyPress={(event) => saveRowDataForInput(event, userId)}
      />
    </div>
  );
};

export default Input;
