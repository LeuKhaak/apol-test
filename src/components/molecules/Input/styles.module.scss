@import "../../../assets/styles/styles.scss";

.inputWrapper {
  border: solid 0.1rem $greyBorderColor;
  padding: 0.5rem;
  input {
    max-width: calc(100% - 0.5rem);
    padding: 0.5rem 0.25rem;
    min-height: 1rem;
    font-weight: 400;
  }
}
