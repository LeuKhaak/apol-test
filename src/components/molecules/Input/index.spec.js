import React from "react";
import { shallow } from "enzyme";
import Input from ".";

let component = null;
const onChange = jest.fn();
const onKeyPress = jest.fn();

const setComponent = (props) => shallow(<Input {...props} />);

describe("Input testing:", () => {
  test("is Input rendered", () => {
    component = setComponent({
      value: "",
      el: ["email", ""],
    });
    expect(component).toMatchSnapshot();
  });

  //test("check onKeyPressFunc", () => { // TypeError: saveRowDataForInput is not a function
  //  component = setComponent({
  //    ind: 0,
  //    el: ["email", ""],
  //    onKeyPress: onKeyPress,
  //  });
  //  component.find("input").simulate("keypress", { key: "Enter", value: "Sk" });
  //  expect(onKeyPress).toHaveBeenCalled();
  //});

  //test("check onChangeFunc", () => { //TypeError: edit is not a function
  //  component = setComponent({
  //    ind: 0,
  //    userId: "gds12l",
  //    el: ["email", ""],
  //    onChange: onChange,
  //  });
  //  component.find("input").simulate("change", { target: { value: "s" } });
  //  expect(onChange).toHaveBeenCalled();
  //});
});
