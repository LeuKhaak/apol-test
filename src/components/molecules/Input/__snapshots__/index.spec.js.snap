// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input testing: is Input rendered 1`] = `
<div>
  <input
    id="undefinedemail"
    name="email"
    onChange={[Function]}
    onKeyPress={[Function]}
    value=""
  />
</div>
`;
